src/libfuncs.rs

```diff
@@ -501,10 +510,20 @@ pub fn increment_builtin_counter<'ctx: 'a, 'a>(
     block: &'ctx Block<'ctx>,
     location: Location<'ctx>,
     value: Value<'ctx, '_>,
+) -> crate::error::Result<Value<'ctx, 'a>> {
+    increment_builtin_counter_by(context, block, location, value, 1)
+}
+
+pub fn increment_builtin_counter_by<'ctx: 'a, 'a>(
+    context: &'ctx Context,
+    block: &'ctx Block<'ctx>,
+    location: Location<'ctx>,
+    value: Value<'ctx, '_>,
+    amount: impl Into<BigInt>,
 ) -> crate::error::Result<Value<'ctx, 'a>> {
     block.append_op_result(arith::addi(
         value,
-        block.const_int(context, location, 1, 64)?,
+        block.const_int(context, location, amount, 64)?,
         location,
     ))
 }
```

Comment:
Why does this exist? (Where do we need to increment a builtin counter by more than one?)

Reply:
In Circuit related libfuncs we need to increment the builtin counter by four. For example: https://github.com/lambdaclass/cairo_native/blob/a54c3f14f4b3b0c524f24a09914d7fd1883b314c/src/libfuncs/circuit.rs#L427-L433

I'm not sure why, but I guess it's related to the fact that each gate contains 4 U96s, so each evaluation increases the builtin by 4.

Reply:
Here is the equivalent CASM compilation: https://github.com/starkware-libs/cairo/blob/aaad921bba52e729dc24ece07fab2edf09ccfa15/crates/cairo-lang-sierra-to-casm/src/invocations/circuit.rs#L260-L262